cppapi/client/devapi_attr.cpp

```diff
@@ -3194,10 +3194,16 @@ bool DeviceAttribute::operator >> (vector<DevState>& datum)
        	 	}
 		}
 		else
-			ret = false;
+			return false;
 	}
 	else
 	{
+		if(d_state_filled)
+		{
+			datum.resize(1);
+			datum[0] = d_state;
+			return true;
+		}
```

Comment:
It is not clear to me why d_state_filled is reset in the example you provided.
It is probably done to simulate the consumption of the data.
If you try to extract twice from the same DeviceData object, at the second invocation check_for_data() will return false because is_empty() will return true, because of this d_state_filled reset to false.
To be consistent, it would probably good to add `d_state_filled = false` in the methods I modified indeed.